test/common/include/alpaka/test/event/EventHostManualTrigger.hpp

```diff
@@ -343,3 +343,346 @@ namespace alpaka
+            {
+                //-----------------------------------------------------------------------------
+                //! \return If the event is not waiting within a stream (not enqueued or already handled).
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto test(
+                    test::event::EventHostManualTriggerCuda const & event)
+                -> bool
+                {
+                    std::lock_guard<std::mutex> lk(event.m_spEventImpl->m_mutex);
+
+                    return event.m_spEventImpl->m_bIsReady;
+                }
+            };
+        }
+    }
+    namespace stream
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //!
+            //#############################################################################
+            template<>
+            struct Enqueue<
+                stream::StreamCudaRtAsync,
+                test::event::EventHostManualTriggerCuda>
+            {
+                //-----------------------------------------------------------------------------
+                //
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto enqueue(
+                    stream::StreamCudaRtAsync & stream,
+                    test::event::EventHostManualTriggerCuda & event)
+                -> void
+                {
+                    ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                    // Copy the shared pointer to ensure that the event implementation is alive as long as it is enqueued.
+                    auto spEventImpl(event.m_spEventImpl);
+
+                    // Setting the event state and enqueuing it has to be atomic.
+                    std::lock_guard<std::mutex> lk(spEventImpl->m_mutex);
+
+                    // The event should not yet be enqueued.
+                    assert(spEventImpl->m_bIsReady);
+
+                    // Set its state to enqueued.
+                    spEventImpl->m_bIsReady = false;
+
+                    // PGI Profiler`s User Guide:
+                    // The following are known issues related to Events and Metrics:
+                    // * In event or metric profiling, kernel launches are blocking. Thus kernels waiting
+                    //   on host updates may hang. This includes synchronization between the host and
+                    //   the device build upon value-based CUDA stream synchronization APIs such as
+                    //   cuStreamWaitValue32() and cuStreamWriteValue32().
+                    ALPAKA_CUDA_DRV_CHECK(
+                        cuStreamWaitValue32(
+                            (CUstream)stream.m_spStreamCudaRtAsyncImpl->m_CudaStream,
+                            (CUdeviceptr)event.m_spEventImpl->m_devMem,
+                            0u,
```

Comment:
Could it be that this muste be `16843009u` because `cuStreamWaitValue32` waits untill `(int32_t)(*addr - value) >= 0`. 
The event internal value (*addr) is set byte wise with cudaMemset to `0x01010101` if `trigger()` is called, before it is zero.

In the current implementation the event finished after queuing because the internal value is set to zero and `0 - 0 >= 0`.

Reply:
I tested the suggested fix, and it is working (expected the later showed re-enqueing test)